### PR TITLE
:seedling: add support for parsing azure devops urls

### PR DESCRIPTION
**Pull request by JamieMagee**
#### What kind of change does this PR introduce?

Initial part of feature #4177.

Mainly to see if the overall feature would be accepted and if I am following the contribution guidelines correctly, before I put a lot of work into the rest of the implementation.

- [x] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the current behavior?

N/A

#### What is the new behavior (if this is a feature change)?**

Azure DevOps URLs can now be parsed

- [x] Tests for the changes have been added (for bug fixes/features)

#### Which issue(s) this PR fixes

Initial part of #4177 

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note
NONE
```

### CODE REVIEW

**Review by raghavkaul — COMMENTED** [review]
Mostly looks good, just one comment.

File: clients/azuredevops/repo.go [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+// Parses input string into repoURL struct
+/*
+ Accepted input string formats are as follows:
+	- "dev.azure.com/<organization:string>/<project:string>/_git/<repository:string>"
+	- "https://dev.azure.com/<organization:string>/<project:string>/_git/<repository:string>"
+*/
+func (r *Repo) parse(input string) error {
+	var t string
+	c := strings.Split(input, "/")
+	if l := len(c); l >= 3 {
```
[the remark this answers is not in the document]
Comment (JamieMagee) [reply clients/azuredevops/repo.go:44]: Yeah, good point. What would you think about omitting it entirely? `url.Parse` and the url path length check are more reliable checks.

[the remark this answers is not in the document]
Comment (raghavkaul) [reply clients/azuredevops/repo.go:44]: Omitting makes sense to me!

[the remark this answers is not in the document]
Comment (JamieMagee) [reply clients/azuredevops/repo.go]: @raghavkaul I removed this check.

**Comment by JamieMagee** [comment]
@raghavkaul If I've addressed all your comments, could you merge this please?

**Review by raghavkaul — APPROVED** [review]

**Pull request merged**